client/src/components/EmojiPicker.js

```diff
@@ -0,0 +1,126 @@
+'use strict';
+
+import React from 'react';
+import ReactEmoji from "react-emoji";
+import { filterEmojis } from "../utils/emojis";
+import 'styles/EmojiPicker.scss';
+
```

Comment:
To simplify the "supported emojis" part. Add the following lines here:
```javascript
import { emojiData } from "../utils/emojilist" // notice the new name

// used by react-emoji, ie. the list of emojis supported by react-emoji
// See: https://github.com/banyan/emoji-annotation-to-unicode/blob/master/emoji-annotation-to-unicode.js for the data structure
import emojisAnnotations from 'emoji-annotation-to-unicode'; 

// Pick the emojis from emojilist that are supported by react-emoji by taking the values (unicode values of the supported emojis)
// Note: we need to filter out value '1f642' as it's a mismatch between react-emoji and emojilist.js and breaks everything if present in the list. Better way would be to do a PR for 'emoji-annotation-to-unicode'...
const supportedEmojis = Object.values(emojisAnnotations).filter((e) => e !== '1f642'); 

// Now we have the list of supported unicode values, let's pick them from the emojilist.js. This will guarantee that we only match with react-emoji supported emojis even if emojilist.js contains them.
const emojiList = _.pickBy(emojiData, (e) => supportedEmojis.indexOf(e.unicode) > -1);
```

Notice also that after this data mangling, we have a clean list from emojilist.js that we can later use (ie. it's not a list of react-emoji's emojified elements).

Bear with me, this will make sense as you read on :)